labconnect/main/auth_routes.py

```diff
@@ -70,36 +75,45 @@ def saml_callback():
     errors = auth.get_errors()
 
     if not errors:
+        registered = True
         user_info = auth.get_attributes()
         # user_id = auth.get_nameid()
 
         data = db.session.execute(db.select(User).where(User.email == "email")).scalar()
 
         # User doesn't exist, create a new user
         if data is None:
-
-            # TODO: add data
-            user = User(
-                # email=email,
-                # first_name=first_name,
-                # last_name=last_name,
-                # preferred_name=json_request_data.get("preferred_name", None),
-                # class_year=class_year,
-            )
-
-            db.session.add(user)
-            db.session.commit()
-
+            registered = False
         # Generate JWT
         # token = create_access_token(identity=[user_id, datetime.now()])
-        code = generate_temporary_code(user_info["email"][0])
+        code = generate_temporary_code(user_info["email"][0], registered)
 
         # Send the JWT to the frontend
         return redirect(f"{current_app.config['FRONTEND_URL']}/callback/?code={code}")
 
     return {"errors": errors}, 500
 
 
+@main_blueprint.post("/register")
+@jwt_required()
+def registerUser():
+    user_id = get_jwt_identity()
+
+    user = db.session.execute(db.select(User).where(User.email == user_id))
```

Comment:
We should be making a new User instead of requesting an existing one, all logic for making a user should be here.